test/gtest/Makefile.am

```diff
@@ -27,8 +27,13 @@ export UCX_HANDLE_ERRORS
 export UCX_LOG_LEVEL
 export UCX_LOG_PRINT_ENABLE
 
-export LSAN_OPTIONS=suppressions=$(top_srcdir)/contrib/lsan.supp
+if HAVE_ASAN
+export LSAN_OPTIONS=suppressions=$(abs_top_srcdir)/contrib/lsan.supp
 export ASAN_OPTIONS=protect_shadow_gap=0
+ASAN_LD_PRELOAD=$(shell ldconfig -p | grep libasan.so | tr ' ' '\n' | grep /)
```

Comment:
1. why need to LD_PRELOAD asan?
2. do we really have to find the full path, maybe just LD_PRELOAD=libasan.so ?

Reply:
1. If pass only `-fsanitize-address` flag on GCC during compiltaion, the following error will appear:
```
ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
```

There are two options to fix it:
- Compile with flag `-static-libasan`, but static ASAN library doesn't exists in default CentOS8 repos. I tried to download it from third-party repo, but this version of static libasan reports some weird errors.
- Set LD_PRELOAD=`path_to_libasan` before running the tests.

2. I can try that and see what happens :)

Reply:
can you pls add comment in the makefile to explain reason for LD_PRELOAD?